galasa-ui/src/components/runs/ArtifactsTab.tsx

```diff
@@ -0,0 +1,308 @@
+"use client";
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import { ArtifactIndexEntry } from '@/generated/galasaapi';
+import { TreeView, TreeNode, InlineLoading, InlineNotification } from '@carbon/react';
+import React, { useEffect, useState } from 'react';
+import styles from '@/styles/Artifacts.module.css';
+import { CloudDownload, Document, Folder, Zip } from '@carbon/icons-react';
+import { downloadArtifactFromServer } from '@/actions/runsAction';
+import { Tile } from '@carbon/react';
+
+interface FileNode {
+  name: string;
+  runId: string;
+  url: string;
+  isFile: true;
+  children: {};
+}
+
+interface FolderNode {
+  name: string;
+  isFile: false;
+  children: { [key: string]: TreeNodeData };
+}
+
+type TreeNodeData = FileNode | FolderNode;
+
+export type DownloadResult =
+  | { contentType: 'application/json'; data: any; size: number; base64: string; }
+  | { contentType: 'text/plain'; data: string; size: number; base64: string; }
+  | { contentType: string; data: string; size: number; base64: string; };
```

Comment:
Would it be easier to use one type rather than three different versions based on what the contentType is?